baxter_collaboration_lib/test/test_ros_thread.cpp

```diff
@@ -0,0 +1,37 @@
+#include <gtest/gtest.h>
+#include "robot_utils/ros_thread.h"
+//#include <pthread.h>
+//#include <typeinfo>
+#include "robot_utils/ros_thread_obj.h"
+
+
+
+class ROSThreadObject: public ROSThread
+{
+   public:
+        int value;
+        void InternalThreadEntry();
+};
+
+ void ROSThreadObject::InternalThreadEntry() {
+        std::cout << "test" << std::endl;
+}; 
+
+TEST(RosThreadTest, testConstructorDefaultValues)
+{
+    ROSThreadObject rt;
+    rt.InternalThreadEntry();
+
+    rt.joinInternalThread();
+    EXPECT_TRUE(rt.startInternalThread());
+    //EXPECT_TRUE(rt.closeInternalThread());
+    EXPECT_TRUE(rt.killInternalThread());
```

Comment:
Are you sure that what you're doing in this thread makes sense? The fact that the tests pass does not mean that what you're testing against makes sense.